xclim/indices/_multivariate.py

```diff
@@ -1477,3 +1662,24 @@ def blowing_snow(
     out = cond.resample(time=freq).sum(dim="time")
     out.attrs["units"] = to_agg_units(out, snd, "count")
     return out
+
+
+def _check_common_sampling_freq(tas, pr):
+    tas_freq = infer_sampling_units(tas)
+    pr_freq = infer_sampling_units(pr)
+    if tas_freq != pr_freq:
+        raise ValidationError(
+            f"The sampling frequencies of the two datasets are different, tas frequency is {tas_freq} whereas pr frequency is {pr_freq}"
+        )
+
+
+def _da_below_per_thrsh(da: xarray.DataArray, da_per: xarray.DataArray):
+    da_per = convert_units_to(da_per, da)
+    thresh = resample_doy(da_per, da)
+    return da < thresh
+
+
+def _da_above_per_thrsh(da: xarray.DataArray, da_per: xarray.DataArray):
+    da_per = convert_units_to(da_per, da)
+    thresh = resample_doy(da_per, da)
+    return da > thresh
```

Comment:
These two functions seem a bit out of place. These operations are typically written into the code of the indices for clarity. If we went with this pattern, I would want to see it applied to the rest of the percentile-based indices as well (ie:`tg/tn/txXXp`).

Reply:
I guess it would be better to do the refactoring of tg/tn/tx..p. in a separated PR anyway. So I will just align this code with the current standard and writes all the operations in each indice.

Reply:
Yes, that would be fine for now. Thank you!